.github/workflows/ros2_pr_test.yaml

```diff
@@ -0,0 +1,24 @@
+---
+name: Test ROS 2 PRs
+
+on:
+  pull_request:
```

Comment:
The original reasons were:
  1. pushes (on master) should result in triggering dockerhub builds and not duplicating the build locally (which is taken care of by the `trigger_devel.yaml` and `trigger_nightly.yaml` workflows).
  1. the `make build` invocation was trying to build `ros2:source` as well which is taking forever to build
  1. building `ros2:nightly` takes a quite some time so we may not want to do it on every single push.
 
Looking at it again it is convenient to get CI feedback without opening a PR so we could add it back.
1. We can easily have this not trigger if the push event happens on the master branch
1. The build of `ros2:source` is now commented out in the Makefile
1. This would be trigger only if the specific subdirectories are modified which doesnt happen very often

So I'll add it back